packages/@webex/plugin-meetings/test/unit/spec/meeting/muteState.js

```diff
@@ -509,3 +509,221 @@ describe('plugin-meetings', () => {
+      },
+    };
+
+    const direction = mediaType === AUDIO ? {sendAudio: true} : {sendVideo: true};
+    muteState = createMuteState(mediaType, meeting, direction, false);
+
+    originalRemoteUpdateAudioVideo = MeetingUtil.remoteUpdateAudioVideo;
+
+    MeetingUtil.remoteUpdateAudioVideo = sinon.stub().resolves(fakeLocus);
+  }
+
+   const setupSpies = (muteState, mediaType) => {
+    applyUnmuteAllowedToTrackSpy = sinon.spy(muteState, 'applyUnmuteAllowedToTrack');
+    muteLocalTrackSpy = sinon.spy(muteState, 'muteLocalTrack');
+    applyClientStateToServerSpy = sinon.spy(muteState, 'applyClientStateToServer');
+
+    setServerMutedSpy = mediaType === AUDIO ? meeting.mediaProperties.audioTrack?.setServerMuted : meeting.mediaProperties.videoTrack?.setServerMuted;
+    setMutedSpy = mediaType === AUDIO ? meeting.mediaProperties.audioTrack?.setMuted : meeting.mediaProperties.videoTrack?.setMuted;
+  };
+
+  const tests = [
+    {mediaType: AUDIO, title: 'audio'},
+    {mediaType: VIDEO, title: 'video'}
+  ];
+
+  tests.forEach(({mediaType, title}) =>
+    describe(title, () => {
+
+      afterEach(() => {
+        MeetingUtil.remoteUpdateAudioVideo = originalRemoteUpdateAudioVideo;
+      });
+
+
+      it('tests handleLocalTrackChange', async () => {
+        setup(mediaType);
+        const spy = sinon.spy(muteState, 'init');
+        muteState.handleLocalTrackChange(meeting);
+        assert.calledOnceWithExactly(spy, meeting);
+      });
+
+      it('tests init when track is undefined', async () => {
+        setup(mediaType, false, false, false);
+        setupSpies(muteState, mediaType);
+
+        muteState.init(meeting);
+
+        assert.calledOnceWithExactly(applyUnmuteAllowedToTrackSpy, meeting);
+        assert.notCalled(muteLocalTrackSpy);
+        assert.notCalled(applyClientStateToServerSpy);
+        assert.isFalse(muteState.state.client.localMute);
+      });
+
+      it('tests init when track muted is true', async () => {
+        setup(mediaType, false, true);
+        setupSpies(muteState, mediaType);
+
+        muteState.init(meeting);
+
+        assert.calledOnceWithExactly(applyUnmuteAllowedToTrackSpy, meeting);
+        assert.notCalled(muteLocalTrackSpy);
+        assert.notCalled(setServerMutedSpy);
+        assert.calledOnceWithExactly(applyClientStateToServerSpy, meeting);
+        assert.isTrue(muteState.state.client.localMute);
+      });
+
+      it('tests init when track muted is false', async () => {
+        setup(mediaType, false, false);
+        setupSpies(muteState, mediaType);
+
+        muteState.init(meeting);
+
+        assert.calledOnceWithExactly(applyUnmuteAllowedToTrackSpy, meeting);
+        assert.notCalled(muteLocalTrackSpy);
+        assert.notCalled(setServerMutedSpy);
+        assert.calledOnceWithExactly(applyClientStateToServerSpy, meeting);
+        assert.isFalse(muteState.state.client.localMute);
+      });
+
+      it('#muteLocalTrack', async () => {
```

Comment:
muteLocalTrack is a private function, I don't think we should be testing it, but the test is actually calling init() instead, so I'm confused, what is being tested here? same for the next test

Reply:
I am trying to specifically test test muteLocalTrack by calling it through muteState.init(meeting) since it is a private method.

Reply:
If we do not want specific its here then I can add checks for serverMted in tests above